### DISCUSSION TITLE
QTabs with multiline option

### DISCUSSION THREAD

**Discussion by hebinet** [quasarframework/quasar#15027] [Ideas / Proposals] [answered]
I have a usecase where I would like to use the tabs bar but the customer don't want the left/right arrows to navigate between the tabs but instead want to always see all tabs.

Something like this
![image](https://user-images.githubusercontent.com/24890467/205062310-cf022454-b1d7-48b4-89ae-6c74c8d8f84f.png)

**Comment by metalsadman** [comment]
well, just use buttons. https://quasar.dev/vue-components/tab-panels#example--basic

**Reply by hebinet** [reply]
Good idea, I tried that already but I want to have the benefits of the tab bar. One property declares which tab is selected, the selected tab gets highlighted and if you switch to another tab, there is a nice little animation for the bottom border moving from the old active to the new active tab.

Yeah, it would be possible to build something like this with buttons but the tab bar does all this already and extending with a new property would be pretty easy.

**Reply by metalsadman** [reply]
having it stacked would screw the animation.

**Reply by hebinet** [reply]
The animation is not the problem. It works good enough in multiline.

How should I use Buttons in an easy way? I can not replace `q-tab` with `q-btn`, this screws everything up.
So in theory, I have to rebuild the complete `q-tabs` and `q-tab` logic with linkting buttons to the selected tab, highlighting the selected and so on.

I tried to modify the `q-tabs` component and it was pretty easy to add some additional classes so that the tabs are wrapping, but this modified `q-tabs` component breaks on every other update.
It would be really nice to have a simple property (maybe `wrap`/`multiline`?) to acomplish this behaviour easily.

On the other side, if I can just use `q-btn`, can you show me some Code how you would do this with them so that it looks and works the same as `q-tabs`?

**Reply by hebinet** [reply]
Ok, I solved it now like this:
```js
<q-tabs
  class="full-width"
  align="left"
  dense
  inverted
  v-model="currentTabIndex"
>
  <div class="flex wrap">
    <template v-for="(tab, index) in navigation" :key="index">
      <q-tab
        :default="index===0"
        :name="`tab-${index}`"
        :label="getTabNavigationName(index)"/>
    </template>
  </div>
</q-tabs>
```
Now it wraps as it should be.

**Comment by metalsadman** [comment, marked as answer]
wrapping the tabs in rows just seem to do it https://codepen.io/metalsadman/pen/MWXLPBK
